src/main/java/org/iot/dsa/dslink/dynamodb/DynamoDBNode.java

```diff
@@ -0,0 +1,469 @@
+package org.iot.dsa.dslink.dynamodb;
+
+import org.iot.dsa.conn.DSBaseConnection;
+import org.iot.dsa.node.DSBool;
+import org.iot.dsa.node.DSFlexEnum;
+import org.iot.dsa.node.DSInfo;
+import org.iot.dsa.node.DSInt;
+import org.iot.dsa.node.DSList;
+import org.iot.dsa.node.DSMap;
+import org.iot.dsa.node.DSString;
+import org.iot.dsa.node.DSValueType;
+import org.iot.dsa.node.action.ActionInvocation;
+import org.iot.dsa.node.action.ActionResult;
+import org.iot.dsa.node.action.ActionSpec.ResultType;
+import org.iot.dsa.node.action.DSAction;
+import org.iot.dsa.node.action.DSActionValues;
+import org.iot.dsa.security.DSPasswordAes128;
+import org.iot.dsa.util.DSException;
+
+import java.util.Collection;
+
+public class DynamoDBNode extends DSBaseConnection {
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Fields
+    ///////////////////////////////////////////////////////////////////////////
+
+    final DSInfo access_key = getInfo(Constants.ACCESSKEY);
+    final DSInfo secret_key = getInfo(Constants.SECRETKEY);
+    final DSInfo region = getInfo(Constants.REGION);
+    final DSInfo endpoint = getInfo(Constants.ENDPOINT);
+
+    //private DSMap parameters;
+    private DynamoDBDSAClient client;
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Constructors
+    ///////////////////////////////////////////////////////////////////////////
+
+    // Nodes must support the public no-arg constructor.  Technically this isn't required
+    // since there are no other constructors...
+    public DynamoDBNode(){
+
+    }
+
+    @Override
+    protected void checkConfig() {
+        if (access_key.getElement().toString().isEmpty()) {
+            configFault("Empty Access Key");
+            throw new IllegalStateException("Empty Access Key");
+        }
+        if (getSecretKey().trim().equals("")) {
+            configFault("Empty Secret Key");
+            throw new IllegalStateException("Empty Secret Key");
+        }
+        if (region.getElement().toString().isEmpty()) {
+            configFault("Empty Region");
+            throw new IllegalStateException("Empty Region");
+        }else {
+            try {
+                com.amazonaws.regions.Regions.fromName(region.getElement().toString());
+            } catch (IllegalArgumentException ex){
+                configFault("Wrong Region");
+                throw new IllegalStateException("Wrong Region");
+            }
+        }
+        if (endpoint.getElement().toString().isEmpty()) {
+            configFault("Empty End-Point");
```

Comment:
Shouldn't it be okay to have an empty endpoint? (Since you handle that case on line 55 of Util.java)

Reply:
In Util we included because we were testing for local DynamoDB. But for clound connection this is mandatory field.

Reply:
I don't think this is always the case. I commented out that check and was able to connect without specifying an endpoint, region was enough.

Reply:
We tested with local dynamodb. commenting out and without specifying endurl we are getting "java.lang.IllegalStateException: The security token included in the request is invalid. (Service: AmazonDynamoDBv2; Status Code: 400; Error Code: UnrecognizedClientException; Request ID: GRR8OEUBBU2EMMH8L4L3MU4A7JVV4KQNSO5AEMVJF66Q9ASUAAJG)
". 

So not sure how it behaves with cloud dynamodb. So is it possible if you have test accesskey and secretkey to share us? So that we test at our end. But we feel there is no harm to keep this code snippet.